msal-device-flow-python/README.md

```diff
@@ -0,0 +1,133 @@
+
+Use these settings in your app registration.
+
+| App registration <br/> setting    | Value for this sample app                                                    | Notes                                                                                              |
+|:---------------------------------:|:-----------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------|
+| **Name**                          | `Python Device Code Flow App`                                                | Suggested value for this sample. <br/> You can change the app name at any time.                    |
+| **Supported account types**       | **Accounts in this organizational directory only (Single tenant)**           | Suggested value for this sample.                                                                   |
+| **Platform type**                 | _None_                                                                       | No redirect URI required; don't select a platform.                                                 |
+| **Allow public client flows**     | **Yes**                                                                      | Required value for this sample.                                                                    |
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
+
+```python
+# Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+"authority": "",
+
+# 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+"client_id": ""
+```
+
+### 3. Install package(s)
+
+To install MSAL libraries:
+
+```bash
+pip install -r requirements.txt
+```
+
+## Run the application
+
+```bash
+python3 app.py
+```
+
+Follow the device code flow instructions that are presented. If everything worked, you should receive a response similar to this:
+
+```console
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users/$entity",
+  "businessPhones": ["+1 (999) 5551001"],
+  "displayName": "Contoso Employee",
+  "givenName": "Contoso",
+  "jobTitle": "Worker",
+  "mail": "cemployee@contoso.com",
+  "mobilePhone": "1 999-555-1001",
+  "officeLocation": "Contoso Plaza/F30",
+  "preferredLanguage": null,
+  "surname": "Employee",
+  "userPrincipalName": "contoso_employee@contoso.com",
+  "id": "e3a49d8b-d849-48eb-9947-37c1f9589812"
+}
+```
+
+## About the code
+
+This Python application will request a user to authenticate via the web browser. Upon successful authentication a request to the Graph API will be made and the response will be displayed.
```

Comment:
This currently describes what the application _does_ rather than the code that enables it.

The intent of this section is to provide a more technical (yet still high-level) overview of how the **code** itself works. Things like which auth library (or libraries) it uses, the primary library types it uses, how the redirects and auth flow goes... that sort of thing.

Reply:
Thanks - I'll work on an update